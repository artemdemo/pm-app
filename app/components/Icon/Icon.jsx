import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const Icon = (props) => {
    const { name, className, onClick } = props;

    if (name === '' && !ENV.production) {
        console.warn('Icon prop `name` couldn\'t be empty');
    }

    const iconClass = classnames(`glyphicon glyphicon-${name}`, className);
    return (
        <span className={iconClass} onClick={onClick} />
    );
};

Icon.propTypes = {
    name: PropTypes.string.isRequired,
    className: PropTypes.string,
    onClick: PropTypes.func,
};

Icon.defaultProps = {
    className: '',
    onClick: null,
};

export default Icon;
