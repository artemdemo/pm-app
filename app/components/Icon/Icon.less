.icon_in-text {
    padding-right: 5px;
}
