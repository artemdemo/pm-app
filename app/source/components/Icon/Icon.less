@import './font-awesome/css/fontawesome-all.css';

.icon_in-text {
    padding-right: 5px;
}
