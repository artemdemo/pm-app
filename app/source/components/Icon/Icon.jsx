import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import './Icon.less';

const Icon = (props) => {
    const { name, inText, className, onClick } = props;

    if (name === '' && !ENV.production) {
        throw new Error('Icon prop `name` couldn\'t be empty');
    }

    const iconClass = classnames(className, {
        glyphicon: true,
        [`glyphicon-${name}`]: true,
        'icon_in-text': inText,
    });
    return (
        <span className={iconClass} onClick={onClick} />
    );
};

Icon.propTypes = {
    name: PropTypes.string.isRequired,
    className: PropTypes.string,
    onClick: PropTypes.func,
    inText: PropTypes.bool,
};

Icon.defaultProps = {
    className: '',
    onClick: null,
    inText: false,
};

export default Icon;
