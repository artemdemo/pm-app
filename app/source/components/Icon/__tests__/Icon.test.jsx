import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../Icon';

describe('<Icon>', () => {
    it('should be rendered with icon name', () => {
        const tree = renderer.create(
            <Icon name='fire' />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should be rendered with className', () => {
        const tree = renderer.create(
            <Icon name='fire' className='some-class' />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should be rendered with "in text" class', () => {
        const tree = renderer.create(
            <Icon name='fire' inText />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should throw if there is no name', () => {
        expect(() => {
            Icon({ name: '' });
        }).toThrow();
    });
});
