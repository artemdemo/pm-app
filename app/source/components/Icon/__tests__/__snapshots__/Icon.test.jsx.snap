// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon> should be rendered with "in text" class 1`] = `
<span
  className="glyphicon glyphicon-fire icon_in-text"
  onClick={null}
/>
`;

exports[`<Icon> should be rendered with className 1`] = `
<span
  className="some-class glyphicon glyphicon-fire"
  onClick={null}
/>
`;

exports[`<Icon> should be rendered with icon name 1`] = `
<span
  className="glyphicon glyphicon-fire"
  onClick={null}
/>
`;
